API/TreeData/TreeDataMock.cs

```diff
@@ -0,0 +1,149 @@
+using System;
+using System.Collections;
+using System.IO;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.Azure.WebJobs;
+using Microsoft.Azure.WebJobs.Extensions.Http;
+using Microsoft.AspNetCore.Http;
+using Microsoft.Extensions.Logging;
+using Newtonsoft.Json;
+
+using MySql.Data;
+using MySql.Data.MySqlClient;
+
+namespace TreeData_CLI
+{
+    public class TreeRequest
+    {
+        public string Subquadrat { get; set; }
+        public string Tag { get; set; }
+        public string StemTag { get; set; }
+        public string SpCode { get; set; }
+        public double DBH { get; set; }
+        public string Codes { get; set; }
+        public string Comments { get; set; }
+        public bool WasDead(ArrayList Trees){

```

Comment:
This function doesn't really belong as a member, since it isn't acting on the properties. To make this function work better with the abstraction, I recommend having a simple member function `IsDead()` which has a simple body `Codes.Contains("dt")` and another function `IsAlive()` which has a body `Codes.Contains("at")`.

Then below you can check for `oldTree.IsDead() && newTree.IsAlive()`

Reply:
Fixed in https://github.com/ForestGeoHack/ForestGEO/pull/12/commits/0543202ea320047257a303f345157882127330c9

Reply:
`IsAlive()` and `IsDead()` should both be defined in the `TreeStorage` entity. Contracts are for service-to-service communication of data, and generally they do not have methods. By defining methods in both the contract and the model, you risk having a function `IsDead` on both classes which produces different behaviors for the same instances.

Reply:
You will run into the trouble the in order to use the methods, you will first need to convert the contract into the model. One way to implement conversion from contract to model is to create a constructor of `TreeStorage` that takes in the `Tree` contract as a parameter.

Reply:
Yeah, that one was a real doozy! I think it's all buttoned up here https://github.com/ForestGeoHack/ForestGEO/pull/12/commits/0cd053264220557007508c3953d0dff6719486b3